src/index.js

```diff
@@ -4,507 +4,609 @@ const codec = require('./codec')
-  opts.port = parseInt(parsed[4])
-  return opts
-}
 
-/**
- * Returns Multiaddr as a human-readable string.
- * For post Node.js v10.0.0.
- * https://nodejs.org/api/deprecations.html#deprecations_dep0079_custom_inspection_function_on_objects_via_inspect
- *
- * @returns {string}
- * @example
- * console.log(Multiaddr('/ip4/127.0.0.1/tcp/4001'))
- * // '<Multiaddr 047f000001060fa1 - /ip4/127.0.0.1/tcp/4001>'
- */
-Multiaddr.prototype[inspect] = function inspectCustom () {
-  return '<Multiaddr ' +
-    uint8ArrayToString(this.bytes, 'base16') + ' - ' +
-    codec.bytesToString(this.bytes) + '>'
-}
-
-/**
- * Returns Multiaddr as a human-readable string.
- * Fallback for pre Node.js v10.0.0.
- * https://nodejs.org/api/deprecations.html#deprecations_dep0079_custom_inspection_function_on_objects_via_inspect
- *
- * @returns {string}
- * @example
- * Multiaddr('/ip4/127.0.0.1/tcp/4001').inspect()
- * // '<Multiaddr 047f000001060fa1 - /ip4/127.0.0.1/tcp/4001>'
- */
-Multiaddr.prototype.inspect = function inspect () {
-  return '<Multiaddr ' +
-    uint8ArrayToString(this.bytes, 'base16') + ' - ' +
-    codec.bytesToString(this.bytes) + '>'
-}
+  /**
+   * Returns Multiaddr as a String
+   *
+   * @example
+   * ```js
+   * new Multiaddr('/ip4/127.0.0.1/tcp/4001').toString()
+   * // '/ip4/127.0.0.1/tcp/4001'
+   * ```
+   */
+  toString () {
+    return codec.bytesToString(this.bytes)
+  }
 
-/**
- * @typedef {object} protocol
- * @property {number} code
- * @property {number} size
- * @property {string} name
- * @property {boolean} [resolvable]
- * @property {boolean} [path]
- */
+  /**
+   * Returns Multiaddr as a JSON encoded object
+   *
+   * @example
+   * ```js
+   * JSON.stringify(new Multiaddr('/ip4/127.0.0.1/tcp/4001'))
+   * // '/ip4/127.0.0.1/tcp/4001'
+   * ```
+   */
+  toJSON () {
+    return this.toString()
+  }
 
-/**
- * Returns the protocols the Multiaddr is defined with, as an array of objects, in
- * left-to-right order. Each object contains the protocol code, protocol name,
- * and the size of its address space in bits.
- * [See list of protocols](https://github.com/multiformats/multiaddr/blob/master/protocols.csv)
- *
- * @returns {protocol[]} protocols - All the protocols the address is composed of
- * @example
- * Multiaddr('/ip4/127.0.0.1/tcp/4001').protos()
- * // [ { code: 4, size: 32, name: 'ip4' },
- * //   { code: 6, size: 16, name: 'tcp' } ]
- */
-Multiaddr.prototype.protos = function protos () {
-  return this.protoCodes().map(code => Object.assign({}, protocols(code)))
-}
+  /**
+   * Returns Multiaddr as a convinient options object to be used with net.createConnection
+   *
+   * @example
+   * ```js
+   * new Multiaddr('/ip4/127.0.0.1/tcp/4001').toOptions()
+   * // { family: 4, host: '127.0.0.1', transport: 'tcp', port: 4001 }
+   * ```
+   */
+  toOptions () {
+    /** @type {MultiaddrObject} */
+    const opts = {}
+    const parsed = this.toString().split('/')
+    opts.family = parsed[1] === 'ip4' ? 4 : 6
```

Comment:
This will mean a breaking change

Reply:
@hugomrdias do we really need to make this change now? This will mean break multiaddr and need to release a major and update it everywhere

Reply:
can you add a breaking change message to the commit?

Reply:
I am thinking if we really want to make a breaking change just for this change and removing the resolve static function. This will mean we need to update all the modules. What do you think?

Reply:
its a breaking change already because the exports changed right ?